src/store/data_layer/data_store.js

```diff
@@ -33,12 +33,168 @@ const state = () => {
       //   }
       // }
     },
-    localStorageChecked: false
+    indexedDBInstance: undefined,
+    remoteSystemNow: undefined
   }
 }
 
 const actions = {
 
+  // Queries backend-api to get "system.now()" timestamp
+  // this is to ensure DeadRequests doesn't improperly
+  // mark intervals as "unavailable."
+  async loadSystemNow (store) {
+    let apiTime = await API.systemtime()
+    // see if we can set the apiTime from querying AWS
+    if (apiTime) {
+      this.commit('dataStore/setSystemNow', { now: Number(apiTime) })
+    } else {
+      // otherwise, fall-back to using locale time
+      this.commit('dataStore/setSystemNow', { now: Date.now() })
+    }
+  },
+
+  // Copies "cache" object to indexedDB for persistent storage
+  async addCacheToIndexedDB (store) {
+    // double-check current store state
+    const db = this.getters['dataStore/DB']
+    if (db === undefined) throw new Error('indexedDB not instantiated')
+    const cache = this.getters['dataStore/cache']
+    if (cache.length === 0) throw new Error('cache object is empty')
+
+    // Wrapping each callback into promise so async works on indexedDB api calls
+    for (let meterId of Object.keys(cache)) {
+      await new Promise((resolve, reject) => {
+        let request = db.transaction('MeterReadings', 'readwrite').objectStore('MeterReadings').put({ 'meterId': meterId, 'meterData': { ...cache[meterId] } })
+        request.onerror = event => reject(event)
+        request.onsuccess = event => resolve(event)
+      })
+        .catch(err => { throw err })
+    }
+  },
+
+  // Adds pairs of intervals which point to non-existant data in the indexDB DeadRequest object store.
+  async addDeadQueries (store, { MissingPairs }) {
+    const db = this.getters['dataStore/DB']
+    if (db === undefined) throw new Error('indexedDB not instantiated')
+
+    MissingPairs.forEach(async pair => {
+      // updates stored dead intervals
+      let [startTime, endTime] = pair
+
+      // check if startTime is at least 3 months old
+      // then we can be reasonably sure that data for that entry
+      // will not be recovered. (Meters cache data for last 3 months)
```

Comment:
[super nitpick] "AcquiSuites" (the data acquisition servers we use) cache the data.

More on them here: http://www.obvius.com/Products/A8812